### ISSUE TITLE
Xorg v1.19.6 breaks Emerald-Window Decorator(Screen Freeze)

### ISSUE THREAD

**Issue by arifcatalyst** [compiz-reloaded/emerald#31]
An update in `Xorg 1.19.6` breaks Emerald-Window-Decorator. It freezes the screen when using `Emerald-Window-decorator` with `Application Switcher` . I can confirm this bug on fresh install of Arch Linux. I am posting since I am still not sure if the bug is in the Video stack(kernel, Xorg, mesa) or `Compiz`

**Comment by arifcatalyst** [comment]
This is the error I get when I start `compiz --replace` 
[log.txt](https://github.com/compiz-reloaded/emerald/files/1595895/log.txt)

**Comment by soreau** [comment]
This seems plausible since the decorator draws the switcher background window, if I remember correctly. Presumably when using gtk-window-decorator, the problem does not happen?

**Comment by arifcatalyst** [comment]
Yes, using `gtk-window-decorator` the problem doesn't happen at all.

**Comment by soreau** [comment]
I am attaching the [fd.o bug report](https://bugs.freedesktop.org/show_bug.cgi?id=104375) here for reference. Have you tried the modesetting driver yet?

I tried in Xephyr and I seem to be able to reproduce the problem but while it is sometimes a freeze and other times a crash in my case, a common pointer seems to be that compiz is spending a lot of time binding. After looking at the code at [texture.c:423](https://github.com/compiz-reloaded/compiz/blob/master/src/texture.c#L423) I noticed it's only if (strictBinding) this case happens (line 421), so I tried the loose binding path and it seems to work around the problem.

TLDR; try starting with `compiz --loose-binding --replace`

**Comment by arifcatalyst** [comment]
No, I haven’t tried with mode setting  driver. Is this a hack or will there be a clean solution ?

**Comment by soreau** [comment]
I don't think you don't need the modesetting driver, just try --loose-binding option.

**Comment by arifcatalyst** [comment]
Can I expect a cleaner hack(Solution). For now, this seems to work.

**Comment by soreau** [comment]
I pinged Michel Dänzer on IRC, we'll have to know what's really going on first.

**Comment by soreau** [comment]
The fact that it doesn't happen with gtk-w-d indicates it might be something that can be fixed in emerald.

**Comment by raveit65** [comment]
Same issue is reported in fedora https://bugzilla.redhat.com/show_bug.cgi?id=1536237

**Comment by raveit65** [comment]
Without testing it looks like downgrading xorg-x11 helps a lot.
https://bugzilla.redhat.com/show_bug.cgi?id=1536237#c4

**Comment by asmca** [comment]
yes I tried to downgrade xorg-X11 on different box. it works.

to trace this issue, we can track the changelog to
xorg-x11-server-Xorg-1.19.6
https://bodhi.fedoraproject.org/updates/FEDORA-2018-4327a9005d

X11 folks might have a better view on this issue.

**Comment by soreau** [comment]
Can anyone give the patch mentioned [here](https://github.com/compiz-reloaded/compiz/issues/106) a try?

**Comment by soreau** [comment]
Also if anyone is in a position to bisect X, it would help to know the exact commit where this began happening.

**Comment by raveit65** [comment]
@soreau 
Sorry, that i didn't found time to test something earlier.
But, i can't reproduce the issue here with f27 and MATE session and 
[rave@mother ~]$ rpm -qa|grep Xorg
xorg-x11-server-Xorg-1.19.6-2.fc27.x86_64
And i am using nvidia-340.xxx for my gt240.
In fedora report the crash happened with using alt+tab with 'application-switcher' or 'static-aplication-switcher inside xfce session.
Are this the correct steps to reproduce the issue?
I don't have xfce installed.
Btw. i forward fedora report to xorg but i didn't get an answer yet.

**Comment by soreau** [comment]
@raveit65 Hi, no worries. Apparently when emerald is running while switching, it triggers the crash.

So far my attempt to bisect the xserver has failed due to not being able to find a good commit. Maybe it's compiz that needs bisecting.

**Comment by soreau** [comment]
I found that building compiz commit 875c9d77 works but git bisect compiz ends with several possible faulty commits. I'm guessing that since some of the relevant decoration code was backported from 0.9, there could be assumptions made about the binding system that need addressing as well.
